plutus-ledger-api/src/v1/value.rs

```diff
@@ -78,12 +145,94 @@ impl Json for CurrencySymbol {
     }
 }
 
+pub(crate) fn currency_symbol(input: &str) -> IResult<&str, CurrencySymbol, VerboseError<&str>> {
+    context(
+        "currency symbol",
+        alt((
```

Comment:
Why not:

```rust
map_res(ledger_bytes, |(LedgerBytes(bytes))| Self::from_bytes(bytes))
```
 
`alt` is kinda expensive in the sense that in the worst case, we have to execute both parsers needlessly

Reply:
good idea, let's roll with this